stats_cache.go

```diff
@@ -0,0 +1,125 @@
+package cedar
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+)
+
+const topN = 10
+const statChanBufferSize = 1000
+
+func newStatsCacheRegistry(ctx context.Context) map[string]*statsCache {
+	registry := map[string]*statsCache{
+		StatsCacheBuildlogger: newStatsCache(StatsCacheBuildlogger),
+		StatsCacheTestResults: newStatsCache(StatsCacheTestResults),
+		StatsCachePerf:        newStatsCache(StatsCachePerf),
+	}
+	for _, r := range registry {
+		go r.startConsumerLoop(ctx)
+	}
+
+	return registry
+}
+
+type Stat struct {
+	Count   int
+	Project string
+	Version string
+	Task    string
```

Comment:
I would change this to `TaskID` to be more specific, since we also have `TaskName`. It may be useful to add `TaskName`.